aten/src/ATen/native/group_norm.cpp

```diff
@@ -31,6 +31,29 @@ std::tuple<Tensor, Tensor, Tensor> native_group_norm(
   const Tensor& gamma = *gamma_maybe_owned;
   const Tensor& beta = c10::value_or_else(beta_opt, [] { return Tensor(); });
 
+  TORCH_CHECK(
+      C % group == 0,
+      "Expected number of channels in input to be divisible by ",
+      "num_groups, but got input of shape ",
+      X.sizes(),
+      " and "
+      "num_groups=",
+      group);
+  TORCH_CHECK(
+      !gamma.defined() || (gamma.dim() == 1 && gamma.numel() == C),
+      "Expected weight to be a vector of size equal to the number of ",
+      "channels in input, but got weight of shape ",
+      gamma.sizes(),
+      " and input of shape ",
+      X.sizes());
+  TORCH_CHECK(
+      !beta.defined() || (beta.dim() == 1 && beta.numel() == C),
+      "Expected bias to be a vector of size equal to the number of ",
+      "channels in input, but got bias of shape ",
+      beta.sizes(),
+      " and input of shape ",
+      X.sizes());
+
```

Comment:
Is it possible to write a `check_group_norm_inputs(...)` function that does this and share it between native_group_norm and group_norm?

We should also leave a comment about why this shouldn't be deleted (someone will come along and inevitably try to delete the additional check). To defend against deletion we probably want to add a test somewhere for aten.ops.native_group_norm that asserts that it checks for these conditions.

Reply:
Moved to helper function, added comment, also added test that expanded weights fails where it should in group norm--I figure that this should be sufficient since we don't exactly care if someone changes what happens in `native_group_norm` if they did it in a way that made expanded weights still correctly fail with incorrect arguments